pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -196,6 +200,13 @@ func (p *DNSProxy) GetRules(endpointID uint16) restore.DNSRules {
 				for _, nid := range cs.GetSelections() {
 					nidIPs := p.LookupIPsBySecID(nid)
 					for _, ip := range nidIPs {
+						// Skip IPs that are allowed but have never been used,
+						// but only if at least one server has been used so far.
+						if len(p.usedServers) > 0 {
```

Comment:
I don't think an additional check is needed here. You can read from an empty (or even nil) map in Go.

Reply:
The intent is to skip the check against `usedServers` if none has been used so for, as otherwise we'd store an empty set of IPs. Could affect a node that:
1) stores DNS rules for restoration before any DNS requests are proxied (no IPs)
2) restarts
3) restores DNS rules without any IPs
4) DNS request is proxied using the restored rules,before the current policy is received from k8s.
5) This DNS request is dropped due to IP not matching the empty set in the restored rule.

Reply:
Ah, makes sense. Would it be worth adding that explanation to the src?